zerver/lib/actions.py

```diff
@@ -2609,6 +2609,22 @@ def truncate_topic(topic):
     # type: (Text) -> Text
     return truncate_content(topic, MAX_SUBJECT_LENGTH, "...")
 
+def do_update_buddy_list(user_profile, buddy_profile, should_add):
+    # type: (UserProfile, str, bool) -> None
+
+    record = BuddyList.objects.get_or_create(user=user_profile, buddy=buddy_profile)
```

Comment:
If should_add is False here, then I don't think we should call `get_or_create` in that case.